pymove/utils/log.py

```diff
@@ -0,0 +1,82 @@
+from IPython.display import display
+from ipywidgets import HTML, IntProgress, VBox
+from tqdm import tqdm
+
+
+def log_progress(sequence, every=None, size=None, desc="Items"):
+    """
+    Make and display a progress bar.
+
+    Parameters
+    ----------
+    sequence : list.
+        Represents a elements sequence.
+
+    every : int, optional, default None.
+        Represents the steps in which the bar will be updated
+
+    size : int, optional, default None.
+        Represents the size/number elements in sequence.
+
+    desc : String, optional, default 'Items'.
+        Represents the description of the operation.
+
+    Returns
+    -------
+    """
+    is_iterator = False
+    if size is None:
+        try:
+            size = len(sequence)
+        except TypeError:
+            is_iterator = True
+    if size is not None:
+        if every is None:
+            if size <= 200:
+                every = 1
+            else:
+                every = int(size / 200)
+    else:
+        assert every is not None, "sequence is iterator, set every"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5513271)